sympy/matrices/common.py

```diff
@@ -784,30 +784,48 @@ def eye(kls, rows, cols=None, **kwargs):
         return klass._eval_eye(rows, cols)
 
     @classmethod
-    def jordan_block(kls, *args, **kwargs):
-        """Returns a Jordan block with the specified size
-        and eigenvalue.  You may call `jordan_block` with
-        two args (size, eigenvalue) or with keyword arguments.
+    def jordan_block(kls, size=None, eigenvalue=None,
+        band='upper', rows=None, cols=None, cls=None, **kwargs):
```

Comment:
If rows and cols are ever deprecated then it would be nice to have cls come after band so we don't have to worry about the order, too:  
->`jordan_block(kls, size=None, eigenvalue=None, band='upper', cls=None, rows=None, cols=None, **kwargs)

Reply:
I think they can be masked under ``kwargs``, so any new feature may be added without worrying about ordering.